docs/upgrade_to_2.1.rst

```diff
@@ -0,0 +1,76 @@
+===========================
+Upgrading to libtorrent 2.1
+===========================
+
+:Author: Arvid Norberg, arvid@libtorrent.org
+
+.. contents:: Table of contents
+  :depth: 2
+  :backlinks: none
+
+In libtorrent 2.1, some parts of the API has changed and some deprecated parts
+have been removed.
+This document summarizes the changes affecting library clients.
+
+C++14 no longer supported
+=========================
+
+libtorrent 2.1 requires at least C++-17. To build with boost build, specify the
+C++ version using the ``cxxstd=17`` build feature (17 is the default).
+
+WebTorrent support
+==================
+
+The major new feature in libtorrent 2.1 is WebTorrent support. WebTorrent
+requires a stun server, it can be configured via the ``webtorrent_stun_server``
+configuration option. It defaults to "stun.l.google.com:19302".
+
+To enable webtorrent support, build with ``webtorrent=on``.
+
+loading and saving torrents
+===========================
+
+The loading and saving of .torrent files and resume data has been simplified and
+responsibilities of classes have become clearer.
+
+	torrent_info
+		Represents the immutable info section of a torrent file. It includes:
```

Comment:
I would also expect `torrent_info` to drop all unrelated functionality, e.g. providing information about effective (changed) filenames will be provided via a separate `file_layout` of torrent, and `torrent_info` will store only the original ones.

Reply:
> I would also expect `torrent_info` to drop all unrelated functionality, e.g. providing information about effective (changed) filenames will be provided via a separate `file_layout` of torrent, and `torrent_info` will store only the original ones.

Ideally there should be no mutators in `torrent_info` interface.

Reply:
Yes, I intend to deprecate those in 2.1